### PR TITLE
[7833] add IdeaViewSet 

### PR DESCRIPTION
**Pull request by goapunk**
**Tasks**
- [x] PR name contains story or task reference
- [x] Documentation (docs and inline)
- [x] Tests (including n+1 and django_assert_num_queries where applicable)
- [x] Changelog

### CODE REVIEW

**Review by philli-m — COMMENTED** [review]
I think this is working well, the api response has everything I think that will be needed for any idea list item and I can make a simplified version for topics. I do have a question though, as I need less information in topics should I make a base view set that can be extended or shall I make one and call it topics as it only needs the name and the comments, and the labels and categories (if added) so it's very basic.
Also did you mean to base it on main?
And did you mean to include the linting experiments? 
I am not sure what else I should test other then the api response, should I be looking at something else?

File: meinberlin/apps/ideas/api.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+from meinberlin.apps.contrib.filters import DefaultsRestFilterSet
+from meinberlin.apps.contrib.filters import OrderingFilterWithDailyRandom
+from meinberlin.apps.contrib.templatetags.contrib_tags import (
+    get_proper_elided_page_range,
+)
+from meinberlin.apps.moderationtasks.models import ModerationTask
+from meinberlin.apps.moderatorfeedback.models import (
+    DEFAULT_CHOICES as moderator_status_default_choices,
+)
+from meinberlin.apps.votes.filters import OwnVotesFilterBackend
```
Comment (philli-m) [inline meinberlin/apps/ideas/api.py:40]: votes are only in budgeting

Comment (goapunk) [reply meinberlin/apps/ideas/api.py]: good catch, fixed

File: meinberlin/apps/ideas/api.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+from .serializers import IdeaSerializer
+
+
+class LocaleInfoMixin:
+    """Add the current locale of the user to the API response"""
+
+    def list(self, request: Request, *args, **kwargs) -> Response:
+        response = super().list(request, args, kwargs)
+        response.data["locale"] = get_language()
+        return response
```
Comment (philli-m) [inline meinberlin/apps/ideas/api.py:49]: I wonder if this would also go int contrib?

**Review by philli-m — APPROVED** [review]
looking good, think everything is there that we need but can improve as we go if not!

File: meinberlin/apps/ideas/api.py (hunk 2 above)
Comment (goapunk) [reply meinberlin/apps/ideas/api.py:49]: @philli-m good point, it definitely could!

**Pull request merged**